### ISSUE TITLE
Boosting fields at build time

### ISSUE THREAD

**Issue by Andargor** [olivernn/lunr.js#267]
As requested in #263 this is a feature request to allow boosting fields at build time, in the form:

```javascript
this.field('title', {boost: 10}); 
```
To expand on desirable behavior, in addition:
- Allow the boosting of exact matches in the field (multi-word for example)
- Allow boosting if the field starts with the term(s)

**Comment by Andargor** [comment]
Example: the search terms "blue whales" would return results in the following order:

- Titles that start with "blue whales"
- Titles that contain "blue whales"
- Titles that have a high score for "blue" and "whales"
- Other results scored with "blue whales" and "blue" and "whales"

**Comment by doesntgolf** [comment]
Reading through #263, I'm also wondering it might also be possible to do field boosting at *query* time as well as build time. (So basically to emulate the behavior that meltuhamy mentioned in the previous issue: `name:pete^10 body:pete`, just in a more straightforward way.) This could leave some room to give end users some flexibility and power in searches (for implementations that exposed the functionality).

**Comment by olivernn** [comment]
I've just released 2.3.0 which includes support for both document and field boosts at build time, as well as the already existing query time boosts.

Documentation/guides to be updated, but the interface looks like this:

```
var idx = lunr(function () {
  this.ref('name')
  this.field('text', { boost: fieldBoost })

  documents.forEach(function (doc) {
    this.add(doc, { boost: documentBoost })
  }, this)
})
```

**Comment by joelanman** [comment]
thanks for this! I was a bit confused by the example, and couldn't find this in the docs, so if anyone else if confused, you can use a number for `fieldBoost` and `documentBoost`, for example:

```
this.field('text', { boost: 2 })
```
